### PR TITLE
Plugins: Add secure JSON fields to plugin setting DTO

### PR DESCRIPTION
**Pull request by wbrowne**
**What this PR does / why we need it**:

The `secureJsonFields` is added to the plugin meta to make it easier for app plugin config editors to detect what fields of the secure json are configured. Before this change, we needed to do:

```tsx
type JsonData = {
  isApiKeySet?: boolean; // need to keep track of what fields are configured in the jsonData
};

type State = {
  isApiKeySet: boolean;
  apiKey: string;
};

interface Props extends PluginConfigPageProps<AppPluginMeta<JsonData>> {}

export const AppConfig = ({ plugin }: Props) => {
  const { enabled, pinned, jsonData } = plugin.meta;
  const [state, setState] = useState<State>({
    apiKey: '',
    isApiKeySet: Boolean(jsonData?.isApiKeySet),
  });

  const onResetApiKey = () =>
    setState({
      ...state,
      apiKey: '',
      isApiKeySet: false,
    });

  const onChangeApiKey = (event: ChangeEvent<HTMLInputElement>) => {
    setState({
      ...state,
      apiKey: event.target.value.trim(),
    });
  };

  return (
    <div>
      <FieldSet label="API Settings">
        <Field label="API Key" description="A secret key for authenticating to our custom API">
          <SecretInput
            width={60}
            value={state?.apiKey}
            isConfigured={state.isApiKeySet}
            placeholder={'Your secret API key'}
            onChange={onChangeApiKey}
            onReset={onResetApiKey}
          />
        </Field>
      </FieldSet>
    </div>
  );
};
```

With this change we can do the following:

```tsx
type State = {
  apiKey: string;
};

interface Props extends PluginConfigPageProps<AppPluginMeta<{}>> {}

export const AppConfig = ({ plugin }: Props) => {
  const { enabled, pinned, jsonData, secureJsonFields } = plugin.meta;
  const [state, setState] = useState<State>({
    apiKey: ''
  });

  const onResetApiKey = () =>
    setState({
      ...state,
      apiKey: ''
    });

  const onChangeApiKey = (event: ChangeEvent<HTMLInputElement>) => {
    setState({
      ...state,
      apiKey: event.target.value.trim(),
    });
  };

  return (
    <div>
      <FieldSet label="API Settings">
        <Field label="API Key" description="A secret key for authenticating to our custom API">
          <SecretInput
            width={60}
            value={state?.apiKey}
            isConfigured={Boolean(secureJsonFields.apiKey)}
            placeholder={'Your secret API key'}
            onChange={onChangeApiKey}
            onReset={onResetApiKey}
          />
        </Field>
      </FieldSet>
    </div>
  );
};
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Added the `secureJsonFields` to the PluginMeta because the `jsonData` and `secureJsonData`(input only) was already added to that type.

But the proper place for it feels like `AppPluginMeta` since it only exists for app plugins. They exists for data sources but only on the instance level and not the plugin level.

### CODE REVIEW

File: pkg/api/dtos/plugins.go [hunk 1]
```diff
@@ -6,18 +6,19 @@ import (
...
+	Id               string                 `json:"id"`
+	Enabled          bool                   `json:"enabled"`
+	Pinned           bool                   `json:"pinned"`
+	Module           string                 `json:"module"`
+	BaseUrl          string                 `json:"baseUrl"`
+	Info             plugins.Info           `json:"info"`
+	Includes         []*plugins.Includes    `json:"includes"`
+	Dependencies     plugins.Dependencies   `json:"dependencies"`
+	JsonData         map[string]interface{} `json:"jsonData"`
+	SecureJsonFields map[string]bool        `json:"secureJsonFields"`
```
Comment (ryantxu) [inline pkg/api/dtos/plugins.go:20]: Is this a real change, or just adding/using type where we currently use `map[string]...`?

I think it is the latter, but correct me if I am wrong

Comment (mckn) [reply pkg/api/dtos/plugins.go:20]: It is a real change. We are adding the `SecureJsonFields` so we can detect what parts of secureJsonData that has been configured.

**Review by mckn — APPROVED** [review]
LGTM! 🙌🏻

File: pkg/api/dtos/plugins.go (hunk 1 above)
Comment (wbrowne) [reply pkg/api/dtos/plugins.go:20]: A real change in the sense that we're passing back `SecureJsonFields` as part of the DTO now similar to datasources

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/api/dtos/plugins.go (hunk 1 above)
Comment (mckn) [reply pkg/api/dtos/plugins.go:20]: This will only be passed back for app plugins, right?

**Pull request merged**